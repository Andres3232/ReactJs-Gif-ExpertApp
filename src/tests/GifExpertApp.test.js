import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Pruebas con el componente <GifExpertApp/>', () => {
    
    test('debe mostrase correctamente ', () => {
        
        const wrapper = shallow(<GifExpertApp />)
        expect( wrapper ).toMatchSnapshot()
    })
    
    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch','Dragon ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length)
    })
    
    
})
