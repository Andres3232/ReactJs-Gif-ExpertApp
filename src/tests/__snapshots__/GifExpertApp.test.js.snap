// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas con el componente <GifExpertApp/> debe mostrase correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
